common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1408,6 +1408,24 @@ public byte toByteExact() {
     throw new NumberFormatException("invalid input syntax for type numeric: '" + this + "'");
   }
 
+  /**
+   * Checks if this UTF8String is valid.
+   *
+   * @return If string represents a valid UTF8 byte string.
+   */
+  public boolean isValid() {
+    return makeValid().equals(this);
+  }
+
+  /**
+   * Returns a validated version of this UTF8String.
+   *
+   * @return A new UTF8String that is a valid UTF8 byte string.
+   */
+  public UTF8String makeValid() {
+    return fromString(toString());
+  }
+
```

Comment:
I presume these are just placeholders, and will be replaced by the implementations in https://github.com/apache/spark/pull/46899, right?

Reply:
yes, exactly - depending on which of these 2 PRs is merged first, a conflict will emerge so we'll update the other PR to ensure that the proper implementation (https://github.com/apache/spark/pull/46899) is used